digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/AwsService.scala

```diff
@@ -0,0 +1,301 @@
+                  }
+                case "Point" => validatePoint(dataset.datasetId, featureId.asInstanceOf[BigInt].longValue(), properties, properties("type").asInstanceOf[String])
+                case _ =>
+                  awsDao.updateFeatureStatus(featureId.asInstanceOf[BigInt].longValue(), "1")
+                  awsDao.updateDatasetStatus(dataset.datasetId, 2)
+              }
+            } catch {
+              case _: Throwable =>
+                awsDao.updateFeatureStatus(featureId.asInstanceOf[BigInt].longValue(), "1")
+                awsDao.updateDatasetStatus(dataset.datasetId, 2)
+            }
+          case None =>
+            featuresWithoutIds += 1
+            awsDao.updateDatasetStatus(dataset.datasetId, 2)
+        }
+      }
+      )
+    }
+    featuresWithoutIds
+  }
+
+  def updateDataset(dataset: Dataset) = {
+    if (awsDao.checkDatasetStatus(dataset.datasetId).toInt != 1) {
+      val assets = dataset.geoJson("features").asInstanceOf[List[Map[String, Any]]]
+      val roadlinks = dataset.roadlinks
+
+      var roadlinksCount = 0
+      assets.foreach(feature => {
+        val featureRoadlinks = roadlinks(roadlinksCount)
+        roadlinksCount += 1
+
+        val properties = feature("properties").asInstanceOf[Map[String, Any]]
+        val featureId = properties.getOrElse("id", 0)
+
+        if (featureId != 0) {
+          val status = awsDao.checkFeatureStatus(featureId.asInstanceOf[BigInt].longValue(), dataset.datasetId)
+          try {
+            if (status == "0") {
+
+              val assetTypeGeometry = feature("geometry").asInstanceOf[Map[String, Any]]("type").asInstanceOf[String]
+              val assetType = properties("type").asInstanceOf[String]
+
+              assetTypeGeometry match {
+                case "LineString" => assetType match {
+                  case "Roadlink" =>
+                    val links = roadLinkService.getRoadLinksAndComplementariesFromVVH(featureRoadlinks.map(_.longValue()).toSet)
+                      .filter(link => link.functionalClass != 99 && link.functionalClass != 7 && link.functionalClass != 8)
+
+                    updateRoadlink(properties, links)
+                }
+                  awsDao.updateFeatureStatus(featureId.asInstanceOf[BigInt].longValue(), "2")
+
+                case "Point" =>
+                  val assetCoordinates = feature("geometry").asInstanceOf[Map[String, Any]]("coordinates").asInstanceOf[List[List[Double]]].head
+                  val link = roadLinkService.getRoadLinkAndComplementaryFromVVH(featureRoadlinks.head.longValue())
+                  updatePoint(properties, link, assetType, assetCoordinates)
+                  awsDao.updateFeatureStatus(featureId.asInstanceOf[BigInt].longValue(), "2")
+              }
+            }
+          } catch {
+            case _: Throwable =>
+              awsDao.updateFeatureStatus(featureId.asInstanceOf[BigInt].longValue(), "1")
```

Comment:
Don't use the catch of the try to treat cases, on this point on the show the specific errors, please.